### PR TITLE
Luisb90/drawer onpush

### PR DESCRIPTION
**Pull request by luisb90**
update drawer component to use OnPush change detection and use type coercion on inputs

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts [hunk 1]
```diff
@@ -7,19 +7,25 @@ import {
...
   ViewEncapsulation,
   OnDestroy,
-  ChangeDetectionStrategy
+  ChangeDetectionStrategy,
+  TemplateRef
 } from '@angular/core';
 import { trigger } from '@angular/animations';
 
 import { drawerTransition } from '../../animations/animations';
+import { coerceNumberProperty, coerceBooleanProperty } from '@angular/cdk/coercion';
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts]: these imports probably belong up with the other angular imports

File: projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts [hunk 2]
```diff
@@ -111,23 +126,20 @@ export class DrawerComponent implements OnDestroy {
...
-  @HostBinding('style.transform')
   transform: string;
 
   /**
    * Drawer width calculation
    *
    * @memberOf DrawerComponent
    */
-  @HostBinding('style.width')
   widthSize: any;
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts]: this as well as `heightSize` could probably be either `string` or `string | number`

Comment (luisb90) [reply projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts]: yeah good call, updated

File: projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts (hunk 1 above)
Comment (luisb90) [reply projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts]: moved it up

**Pull request merged**